add update script with github actions

omarcresp opened the pull request:
This pull request introduces a new GitHub Action workflow for automating updates to the Zen Browser, along with related changes to support this automation. The primary changes include the addition of the workflow file, modifications to the `flake.nix` file, and a new script for updating the Zen Browser version and hashes.

Automation and workflow:

* [`.github/workflows/zen-update.yml`](diffhunk://#diff-2c3851a5459132fee03426d6f5d5b2a1ab29e93bcfdfd8ca010d70e6a71e51edR1-R34): Added a new GitHub Action workflow to automate the update process for Zen Browser. This workflow includes steps for checking out the repository, installing Nix, setting up a Nix cache, running the update script, and committing changes.

Supporting changes:

* [`update-zen-browser.sh`](diffhunk://#diff-3f2d827bef46de2ee4ccfe5d8fe58c8840d6a4b7f533e07c9b395155e5f54da6R1-R22): Added a new script to check for the latest Zen Browser release, update the version in `flake.nix`, and update the corresponding SHA256 hashes.
* [`flake.nix`](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0L13-R17): Modified the format for specifying the download URLs and SHA256 hashes to simplify updates by the new script.
* [`version`](diffhunk://#diff-5ca4f3850ccc331aaf8a257d6086e526a3b42a63e18cb11d020847985b31d188R1): Added a file to store the current version of Zen Browser being used, which is referenced by the update script.

omarcresp committed:
add update script with github actions

omarcresp committed:
stop nix install if theres no new version build required

ch4og reviewed (approved):
LGTM, the only thing I don't really like is creation of version file, version can be read from flake.nix without creating a version file

smth like `grep -oP 'version\s*=\s*"\K[^"]+' flake.nix`

ch4og commented on update-zen-browser.sh:
```diff
@@ -1,25 +1,21 @@
 
-  # Modiyf with sed the nix file
-  sed -i "s/version = \".*\"/version = \"$upstream\"/" ./flake.nix
+# Modiyf with sed the nix file
```
typo in word "Modify"

omarcresp committed:
remove need of version file & fix typo

omarcresp commented:
> LGTM, the only thing I don't really like is creation of version file, version can be read from flake.nix without creating a version file
> 
> smth like `grep -oP 'version\s*=\s*"\K[^"]+' flake.nix`

updated, removed the version file and replaced with the regex

chakibchemso reviewed (approved):
great contribution!

luisnquin commented:
This is an important contribution and it seems the current owner is no longer interested in maintaining it so I just created a fork of this project for those who really want more activity.

You can find it at https://github.com/0xc000022070/zen-browser-flake.
